soroban-sdk/src/address.rs

```diff
@@ -285,6 +293,32 @@ impl Address {
         }
     }
 
+    /// Returns the executable type of this address, if any.
+    ///
+    /// For Wasm contracts, this also returns the hash of the contract code.
+    /// Otherwise, this just returns which kind of 'built-in' executable this is
+    /// (StellarAsset or Account).
+    pub fn executable(&self) -> Option<Executable> {
+        let executable_val: Val =
+            Env::get_address_executable(&self.env, self.obj).unwrap_infallible();
+        if executable_val.is_void() {
+            return None;
+        }
+        Some(executable_val.into_val(&self.env))
```

Comment:
Does the following not work?
```suggestion
    pub fn executable(&self) -> Option<Executable> {
        let executable_val: Val =
            Env::get_address_executable(&self.env, self.obj).unwrap_infallible();
        executable_val.into_val(&self.env)
```

Reply:
Option<T> is implemented as void -> None, otherwise T -> Som.